src/main/java/serene/Serene.java

```diff
@@ -0,0 +1,402 @@
+        case "bye":
+            operationState = DONE;
+            break;
+        case "list":
+            printTaskList();
+            break;
+        case "mark":
+            markTaskDone(responsePartition);
+            break;
+        case "unmark":
+            markTaskNotDone(responsePartition);
+            break;
+        case "delete":
+            removeTask(responsePartition);
+            break;
+        default:
+            addTask(userInput);
+        }
+        return operationState;
+    }
+
+    private static void printTaskList() {
+        System.out.println(PARTITION_LINE);
+        System.out.println("Here is your task list:");
+        int i = 1;
+        for (Task task : taskList) {
+            System.out.println((i) + "." + task);
+            i++;
+        }
+        System.out.println(PARTITION_LINE);
+    }
+
+    private static void markTaskDone(String[] userInput) {
+        try {
+            // Extract index of task to mark
+            int taskIndex = validateIndex(userInput);
+            if (taskIndex == ERROR_CODE) {
+                return;
+            }
+            // Checking if task has not already been marked
+            if (!taskList.get(taskIndex).isDone()) {
+                taskList.get(taskIndex).markDone();
+                printWithPartition("Good job~ This task is now done:" + System.lineSeparator() +
+                        taskList.get(taskIndex));
+            }
+            else {
+                printWithPartition("Huh? Didn't you complete this already?" + System.lineSeparator() +
+                        taskList.get(taskIndex));
+            }
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            printWithPartition(INVALID_NUM_ERROR_MESSAGE);
+        }
+    }
+
+    private static void markTaskNotDone(String[] userInput) {
+        try {
+            // Extract index of task to unmark
+            int taskIndex = validateIndex(userInput);
+            if (taskIndex == ERROR_CODE) {
+                return;
+            }
+            // Checking if task has already been marked
+            if (taskList.get(taskIndex).isDone()) {
+                taskList.get(taskIndex).markNotDone();
+                printWithPartition("Sigh. Here we go again:" + System.lineSeparator() +
+                        taskList.get(taskIndex));
+            }
+            else {
+                printWithPartition("Bruh. You never completed this in the first place." + System.lineSeparator() +
+                        taskList.get(taskIndex));
+            }
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            printWithPartition(INVALID_NUM_ERROR_MESSAGE);
+        }
+    }
+
+    private static void removeTask(String[] userInput) {
+        try {
+            int taskIndex = validateIndex(userInput);
+            if (taskIndex == ERROR_CODE) {
+                return;
+            }
+            if (taskCount == 2) {
```

Comment:
`2` is a [magic number](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) which you could replace with a named constant to improve code quality.